linkcheck/checker/httpurl.py

```diff
@@ -84,8 +84,6 @@ def content_allows_robots (self):
         handler = linkparse.MetaRobotsFinder()
         parser = htmlsax.parser(handler)
         handler.parser = parser
-        if self.charset:
-            parser.encoding = self.charset
```

Comment:
I have doubts about this entire commit.

If we have charset information from the Content-Type header, there should be a way to force the HTML parser to use it.  (The spec requires that if the document itself specifies charset X in a `<meta>` tag, but the server specifies charset Y in the Content-Type header, then the parser must use charset Y.)

Reply:
It's straightforward to implement (add a get_content() method to HttpUrl that takes the encoding from requests and tells Beautiful Soup to use it). Harder to test all the permutations.